evap/grades/views.py

```diff
@@ -117,29 +117,19 @@ def upload_grades(request, semester_id, course_id):
 
 
 @grade_publisher_required
-def toggle_no_grades(request, semester_id, course_id):
-    semester = get_object_or_404(Semester, id=semester_id)
+def toggle_no_grades(request):
+    course_id = request.POST.get("course_id")
     course = get_object_or_404(Course, id=course_id)
```

Comment:
what do you mean by resolving URLs in javascript?

what i mean is basically leave the urls.py and the URL creation in the templates unchanged, and in the view, leaving the view parameters as they are. so, continue doing

in the template:
```
{% url "grades:toggle_no_grades" semester.id course.id %}
```
and pass this whole url into the javascript function. the url tag can also do `as variable_name` if that helps.

urls.py
```
url(r"^semester/(\d+)/course/(\d+)/delete/(\d+)$", delete_grades, name="delete_grades"),
```

in the view
```
def toggle_no_grades(request, semester_id, course_id):
    semester = get_object_or_404(Semester, id=semester_id)
    course = get_object_or_404(Course, id=course_id)
    blahbla
```
just like it was before.